learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+
+```{r}
+#| label: "plot-insurer"
+readmission_collapsed %>%
+  ggplot(aes(x = insurer)) +
+  geom_bar()
+```
+
+While payment information on most patients is missing, most patients in this data are covered under medicare.
+
+::: callout-tip
+The payment method is one way in which societal unfairness may be reflected in the source data besides the variables on protected groups themselves. Medicaid coverage is only available to people below a certain income, and many self-pay patients do not have medical insurance because they cannot afford it. Relatedly, poverty rates differ drastically among racial groups in the U.S.
+:::
+
+```{r}
+#| label: "plot-n-variables"
+#| message: false
+#| warning: false
+readmission_collapsed %>%
+  pivot_longer(starts_with("n_")) %>%
+  ggplot(aes(x = value)) +
+  geom_histogram() +
+  facet_wrap(vars(name), scales = "free_x")
+```
+
+For many patients, this was their first inpatient visit to this hospital system. During their stay, many patients receive 10-20 medications and experience several procedures.
+
+With a sense for the distributions of variables in this dataset, we can move on to splitting data up for modeling.
+
+## Resampling Data
+
+First, splitting data into training and testing:
+
+```{r}
+#| label: "split"
+set.seed(1)
+readmission_splits <- initial_split(readmission_collapsed)
+readmission_train <-  training(readmission_splits)
+readmission_test <-   testing(readmission_splits)
+```
+
+We'll set the `r format(nrow(readmission_test), big.mark = ",")`-row `readmission_test` test set to the side for the remainder of the analysis until we compute a final estimate of our performance on the chosen model. Splitting the `r format(nrow(readmission_train), big.mark = ",")` rows of the training data into 10 resamples:
+
+```{r}
+#| label: "resample"
+readmission_folds <- vfold_cv(readmission_train)
+
+readmission_folds
+```
+
+Each split contains an analysis and assessment set: one for model fitting and the other for evaluation. Averaging performance estimates across resamples will give us a sense for how well the model performs on data it hasn't yet seen.
+
+## Training and Evaluating Models
+
+We'll define a diverse set of models and pre-processing strategies and then evaluate them against our resamples.
+
+### Model Workflows
+
+We'll first define a basic recipe that first sets a factor level for missing values and then centers and scales numeric data:
+
+```{r}
+#| label: "recipe-basic"
+recipe_basic <-
+  recipe(readmitted ~ ., data = readmission) %>%
+  step_unknown(all_nominal_predictors()) %>%
+  step_YeoJohnson(all_numeric_predictors()) %>%
+  step_normalize(all_numeric_predictors()) %>%
+  step_dummy(all_nominal_predictors())
+```
+
+The other preprocessor that we'll try encodes the age as a numeric variable rather than the bins as in the source data:
+
+```{r}
+#| label: "recipe-age"
```

Comment:
I would add some comments here; it would be intimidating for folks that are not R experts.

Reply:
Also, I would add a better name for the x-axis

Reply:
If you are trying to show the relationship between the metrics (i.e. as ROC increses, X decreases), I would use a scatterplot matrix instead.

Reply:
What are the comments about a plot referring to? Is the comment about adding comments referring to that plot as well or `"recipe-age"`?